operators/src/pro/ml/xgboost.rs

```diff
@@ -0,0 +1,1105 @@
+                },
+                GridOrEmpty::from(Grid2D::new([5, 5].into(), t2).unwrap()),
+            ),
+        ];
+
+        MockRasterSource {
+            params: MockRasterSourceParams {
+                data: raster_tiles,
+                result_descriptor: RasterResultDescriptor {
+                    data_type: RasterDataType::U8,
+                    spatial_reference: SpatialReference::epsg_4326().into(),
+                    measurement: Measurement::Unitless,
+                    time: None,
+                    bbox: None,
+                    resolution: None,
+                },
+            },
+        }
+        .boxed()
+    }
+
+    fn make_raster(tile: Vec<i32>) -> Box<dyn RasterOperator> {
+        // green raster:
+        // 255, 255,   0,   0,   0
+        // 255, 255,   0,   0,   0
+        //   0,   0,   0,   0,   0
+        //   0,   0,   0, 255, 255
+        //   0,   0,   0, 255, 255
+        let raster_tiles = vec![RasterTile2D::<i32>::new_with_tile_info(
+            TimeInterval::new_unchecked(0, 1),
+            TileInformation {
+                global_tile_position: [-1, 0].into(),
+                tile_size_in_pixels: [5, 5].into(),
+                global_geo_transform: TestDefault::test_default(),
+            },
+            GridOrEmpty::from(Grid2D::new([5, 5].into(), tile).unwrap()),
+        )];
+
+        MockRasterSource {
+            params: MockRasterSourceParams {
+                data: raster_tiles,
+                result_descriptor: RasterResultDescriptor {
+                    data_type: RasterDataType::U8,
+                    spatial_reference: SpatialReference::epsg_4326().into(),
+                    measurement: Measurement::Unitless,
+                    time: None,
+                    bbox: None,
+                    resolution: None,
+                },
+            },
+        }
+        .boxed()
+    }
+
+    // Just a helper method to extract xgboost matrix generation code.
+    fn make_xg_matrix(data_arr_2d: &ArrayBase<OwnedRepr<f32>, Dim<[usize; 2]>>) -> DMatrix {
+        // define information needed for xgboost
+        let strides_ax_0 = data_arr_2d.strides()[0] as usize;
+        let strides_ax_1 = data_arr_2d.strides()[1] as usize;
+        let byte_size_ax_0 = std::mem::size_of::<f32>() * strides_ax_0;
+        let byte_size_ax_1 = std::mem::size_of::<f32>() * strides_ax_1;
+
+        // get xgboost style matrices
+        let xg_matrix = DMatrix::from_col_major_f32(
+            data_arr_2d.as_slice_memory_order().unwrap(),
+            byte_size_ax_0,
+            byte_size_ax_1,
+            25,
+            3,
+            -1,
+            f32::NAN,
+        )
+        .unwrap();
+
+        xg_matrix
+    }
+
+    /// This test is used to:
+    /// a) train a xgboost model
+    /// b) predict unseen data
+    /// c) check, if the predictions are correct
+    /// This test only checks, if the xgboost library works correctly
```

Comment:
idk if we want to test the correctness of libraries we use. Maybe move this test to your library instead?

Reply:
I removed the test. My reasoning was to use this test in order to verify if the library works in the context of tile data.